apps/aecore/src/aec_peer_connection.erl

```diff
@@ -954,22 +954,33 @@ handle_new_micro_block(S, Msg) ->
                 Header = aec_blocks:to_header(MicroBlock),
                 {ok, HH} = aec_headers:hash_header(Header),
                 epoch_sync:debug("Got new block: ~s", [pp(HH)]),
+                %% in the full micro block case - conductor will do the necessary checks
                 aec_conductor:post_block(MicroBlock);
             true ->
                 {ok, #{ header := Header, tx_hashes := TxHashes }} =
                     deserialize_light_micro_block(maps:get(micro_block, Msg)),
-                case get_micro_block_txs(TxHashes) of
-                    {all, Txs} ->
-                        MicroBlock = aec_blocks:new_micro_from_header(Header, Txs),
+                %% Before assembling the block, check if it is known, and valid
```

Comment:
I see the original description of https://www.pivotaltracker.com/n/projects/2124891/stories/160093084 was:
> The process representing a connection with a peer inserts a micro-block expansion internal message keyed by header hash {expand, Hash}. This request is queued following (unsolicited) request (not response) by peer after only structural parsing (not sanitized).

I see this PR addresses this by performing stricter checks for gossiped micro blocks: only microblocks at recent enough height trigger this process, and even those actors able to do so (for signature) are bound to timestamp. Ok.